src/olympia/activity/tests/test_views.py

```diff
@@ -187,3 +195,65 @@ def _set_tested_url(self, pk=None, version_pk=None, addon_pk=None):
         self.url = reverse('version-reviewnotes-list', kwargs={
             'addon_pk': addon_pk or self.addon.pk,
             'version_pk': version_pk or self.version.pk})
+
+
+@override_settings(ALLOWED_CLIENTS_EMAIL_API=['10.10.10.10'])
+@override_settings(POSTFIX_AUTH_TOKEN='something')
+class TestEmailApi(TestCase):
+
+    def get_request(self, data=None, user=None):
+        data = data or {}
+        datastr = json.dumps(data)
+        req = req_factory_factory(reverse('post-email-api'), user=user)
+        req.META['REMOTE_ADDR'] = '10.10.10.10'
+        req.META['HTTP_POSTFIX_AUTH_TOKEN'] = 'something'
+        req.META['CONTENT_LENGTH'] = len(datastr)
+        req.META['CONTENT_TYPE'] = 'application/json'
+        req.POST = data
+        req.method = 'POST'
+        req._stream = StringIO.StringIO(datastr)
+        return req
+
+    def test_basic(self):
+        user = user_factory()
+        self.grant_permission(user, '*:*')
+        addon = addon_factory()
+        req = self.get_request(data={'body': open(sample_email).read()},
+                               user=user)
+
+        ActivityLogToken.objects.create(
+            user=user, version=addon.latest_version,
+            uuid='5a0b8a83d501412589cc5d562334b46b')
+
+        res = post_email(req)
+        assert res.status_code == 201
+        logs = ActivityLog.objects.for_addons(addon)
+        assert logs.count() == 1
+        assert logs.get(action=amo.LOG.REVIEWER_REPLY_VERSION.id)
+
+    def test_allowed(self):
+        assert EmailCreationPermission().has_permission(
+            self.get_request(), None)
+
+    def test_ip_denied(self):
+        req = self.get_request()
+        req.META['REMOTE_ADDR'] = '10.10.10.1'
+        assert not EmailCreationPermission().has_permission(
+            req, None)
+
+    def test_token_denied(self):
+        req = self.get_request()
+        req.META['HTTP_POSTFIX_AUTH_TOKEN'] = 'somethingwrong'
+        assert not EmailCreationPermission().has_permission(req, None)
+
+    @mock.patch('olympia.activity.tasks.process_email.apply_async')
+    def test_successful(self, _mock):
+        req = self.get_request({'body': 'something'})
+        res = post_email(req)
+        _mock.assert_called_with(('something',))
+        assert res.status_code == 201
+
+    def test_bad_request(self):
+        """Test with no email body."""
+        res = post_email(self.get_request())
+        assert res.status_code == 400
```

Comment:
Could you add a test hitting the endpoint without the `POSTFIX_AUTH_TOKEN` header?